src/SdfEntityCreator.cc

```diff
@@ -788,8 +794,44 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Visual *_visual)
   // \todo(louise) Populate with default material if undefined
   if (_visual->Material())
   {
+    sdf::Material visualMaterial = *_visual->Material();
+    if (!_visual->Material()->ScriptUri().empty())
+    {
+      gzwarn << "Gazebo does not support Ogre material scripts. See " <<
+      "https://gazebosim.org/api/sim/8/migrationsdf.html#:~:text=Materials " <<
+      "for details." << std::endl;
+      std::string scriptUri = visualMaterial.ScriptUri();
+      if (scriptUri != "file://media/materials/scripts/gazebo.material") {
+        gzwarn << "Custom material scripts are not supported."
+          << std::endl;
+      }
+    }
+    if (!_visual->Material()->ScriptName().empty())
+    {
+      std::string scriptName = visualMaterial.ScriptName();
+
+      if ((scriptName.find("Gazebo/") == 0u))
+      {
+        gzwarn << "Using an internal gazebo.material to parse "
+          << scriptName << std::endl;
+        MaterialParser::MaterialValues parsed =
+          this->dataPtr->materialParser.GetMaterialValues(scriptName);
+
+        if(!parsed.ambient.has_value()) {
```

Comment:
Looks like this will print a warning if ambient is not defined. Is that sufficient to know if the material wasn't recognized? Are there materials that define just diffuse or specular?

Reply:
There aren't any solid colors that don't have ambient but I'm updating the parser logic (as you mentioned below) to add the warning there itself